targets/tinyfpga_bx/internal.py

```diff
@@ -0,0 +1,177 @@
+import sys
+import struct
+import os.path
+import argparse
+
+from migen import *
+from migen.genlib.resetsync import AsyncResetSynchronizer
+
+from litex.build.generic_platform import ConstraintError
+from litex.build.generic_platform import Pins, Subsignal, IOStandard, Misc
+from litex.soc.integration.soc_core import *
+from litex.soc.integration.builder import *
+
+from gateware import info
+#from gateware import cas
+from gateware import spi_flash
+
+
+serial =  [
+    ("serial", 0,
+        Subsignal("rx", Pins("GPIO:0")), # Pin 1 - A2 - Silkscreen 1
+        Subsignal("tx", Pins("GPIO:1")), # Pin 2 - A1 - Silkscreen 2
+        IOStandard("LVCMOS33")
+    )
+]
+
+reset_button =  [
+    # Reset shorts these two pins together
+    # Pin 12 - Silkscreen 12
+    ("reset", 0, Pins("GPIO:11"), IOStandard("LVCMOS33"), Misc("PullUp")),
+    # Pin 13 - Silkscreen 13
+    ("reset_out", 0, Pins("GPIO:12"), IOStandard("LVCMOS33")),
+]
+
+
+class _CRG(Module):
```

Comment:
This should go into a `crg.py` file to be consistent with the other platforms?

Reply:
The implementation of `_CRG` class in `targets/tinyfpga_bx/internal.py` is a bit different  than the one in `targets/tinyfpga_bx/base.py`.

It adds two reset signals, connects `user_led` and initializes special of type `SB_PLL40_CORE` - are those changes necessary and should be used in `base` target as well, @mithro? It comes from your commit https://github.com/timvideos/litex-buildenv/commit/574809ef1814acdffa23d28e0def6ceefd2201cf.